File: backend/src/main/java/com/pgvaale/backend/dto/UserTiffinDTO.java
package com.pgvaale.backend.dto;

import com.pgvaale.backend.entity.UserTiffin;
import lombok.*;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserTiffinDTO {
    private Long id;
    private Long userId;
    private Long tiffinId;
    private String userName;
    private String tiffinName;
    private String status;
    private LocalDateTime assignedDateTime;
    private LocalDateTime deletionDateTime;
    
    public static UserTiffinDTO fromEntity(UserTiffin userTiffin) {
        return UserTiffinDTO.builder()
                .id(userTiffin.getId())
                .userId(userTiffin.getUser().getId())
                .tiffinId(userTiffin.getTiffin().getId())
                .userName(userTiffin.getUser().getName())
                .tiffinName(userTiffin.getTiffin().getName())
                .status(userTiffin.getStatus().name())
                .assignedDateTime(userTiffin.getAssignedDateTime())
                .deletionDateTime(userTiffin.getDeletionDateTime())
                .build();
    }
} 

File: PGVaaleDotNetBackend/DTOs/UserTiffinDTO.cs
using PGVaaleDotNetBackend.Entities;

namespace PGVaaleDotNetBackend.DTOs
{
    public class UserTiffinDTO
    {
        public long Id { get; set; }
        public long UserId { get; set; }
        public long TiffinId { get; set; }
        public DateTime AssignedDateTime { get; set; }
        public DateTime? DeletionDateTime { get; set; }
        public UserTiffin.RequestStatus Status { get; set; }
        public string UserName { get; set; } = string.Empty;
        public string TiffinName { get; set; } = string.Empty;

        public static UserTiffinDTO FromEntity(UserTiffin entity)
        {
            return new UserTiffinDTO
            {
                Id = entity.Id,
                UserId = entity.UserId,
                TiffinId = entity.TiffinId,
                AssignedDateTime = entity.AssignedDateTime,
                DeletionDateTime = entity.DeletionDateTime,
                Status = entity.Status,
                UserName = entity.User?.Name ?? string.Empty,
                TiffinName = entity.Tiffin?.Name ?? string.Empty
            };
        }
    }
}